manual/manual/tutorials/lablexamples.etex

```diff
@@ -69,15 +69,6 @@ let hline ~x:x1 ~x:x2 ~y = (x1, x2, y);;
 hline ~x:3 ~y:2 ~x:5;;
 \end{caml_example}
 
-As an exception to the above parameter matching rules, if an
-application is total (omitting all optional arguments), labels may be
-omitted.
-In practice, many applications are total, so that labels can often be
-omitted.
-\begin{caml_example}{toplevel}
-f 3 2;;
-ListLabels.map succ [1;2;3];;
-\end{caml_example}
```

Comment:
I don't agree with removing an explanation for a behavior that still exists and is used in some code bases. 

We could comment on the fact that unlabelled total application are nowadays discouraged.

Also the paragraph below "But beware" is still part of this explanation.

Reply:
This is a tutorial for newcomers, not the reference documentation on labels. I don't think that historical remarks are relevant in the tutorial, they obscure the flow, no. (I'll remove the "But beware" stuff below, thanks.). Do you think that the reference sections on labels should be enriched to comment on full applications and they disrecommendation?

Reply:
I located a part of the reference manual that also mentions labels-omitted applications, I will mention the change there.

Reply:
There is no reference section on labels. Thus functionally, this section in the tutorial is by far the most accessible and exhaustive documentation that most people will find. And labels-omitted applications are still part of current code bases, you cannot relegate it to history as soon as we switch the compiler set of warning.

Yes, there is a mention to label-omitted applications in the language reference part of the manual, but it is hard to find without knowing that label-omitted applications exists.

If you don't want to break the flow of the section, why not move this explanation to a subsection at its end with a warning that most reader can skip this subsection?

Reply:
> There is no reference section on labels. 

This is incorrect. [Link](https://caml.inria.fr/pub/docs/manual-ocaml/expr.html#sss:expr-functions-application).
(You mention this section later in your message.)

> Thus functionally, this section in the tutorial is by far the most accessible and exhaustive documentation that most people will find.

It is also short and readable, and I want to keep it that way. Again, historical remarks have no place in a short tutorial. If you want to have a better reference section on labels, feel free to send PRs to improve it.

(I spent time trying to rewrite this sub-sub-section of the tutorial to try to explain the issue and make it better, and the result was long and verbose and out of place, so I got rid of it.)

I reformulated this PR again to add a more extensive discussion of labels-omitted in the reference sections on Warnings, with a forward reference to this explanation from the reference section.

Reply:
Once again historical does not mean "yesterday from the point of view of compiler developers". But the section on warning explanation does seem discoverable enough.